autoload/neomake.vim

```diff
@@ -247,11 +249,20 @@ function! s:AddExprCallback(maker) abort
             call Func(entry)
         end
 
-        if entry.valid && !file_mode
+        if !entry.valid
+            if get(a:maker, "remove_invalid_entries", 0)
```

Comment:
I think the default should be 1, since the most common use case is syntax checking.